### PR TITLE
Multistore - International > Taxes > Taxes - Add checkboxes

### LINKED ISSUE #19325: Multistore - International > Taxes > Taxes - Add specific settings dropdown
As a merchant, I want to be able to see if a parameter has a different value for another store when I am in "all shops" context or in a group context, so I can know to which stores the change I am making will be applied

[Specs specific settings dropdown](https://github.com/PrestaShop/prestashop-specs/blob/9783996765a46d3d7197e4f7cd7098ffb7e37ff2/content/1.7/back-office/multistoregeneralspecs.md#specific-settings-drop-down-only-from-178)

### LINKED ISSUE #19373: Multistore - International > Taxes > Taxes - Add checkboxes
As a merchant, I want to be able to select one or many fields / toggles when I am in a specific shop context or in a group context, so I can apply a different configuration for the selected store or group and see which parameters have different value(s) for the selected store or group.

[Specs checkboxes](https://github.com/PrestaShop/prestashop-specs/blob/master/back-office/multistoregeneralspecs.md#checkboxes)

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | As a merchant, I want to be able to select one or many fields / toggles when I am in a specific shop context or in a group context, so I can apply a different configuration for the selected store or group and see which parameters have different value(s) for the selected store or group.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/19373 Fixes https://github.com/PrestaShop/PrestaShop/issues/19325
| How to test?      | See issue
| Possible impacts? | 

:information_source:  **Informations BC Breaks**

- `TaxOptionsConfiguration` construtor changes
- New agrument `$configuration` in `private function updateEcotax($isEnabled, $configuration)`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27525)

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Edit : Fail with reviewvable.io

**Review by matthieu-rolland — DISMISSED** [review]

File: tests/Unit/Core/Tax/TaxOptionsConfigurationTest.php [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        $this->mockConfiguration = $this->createConfigurationMock();
+        $this->mockShopConfiguration = $this->createShopContextMock();
+        $this->mockMultistoreFeature = $this->createMultistoreFeatureMock();
+        $this->mockProductEcotaxResetter = $this->createProductEcotaxResetterMock();
+    }
+
+    /**
+     * @return ProductEcotaxResetterInterface
+     */
+    protected function createProductEcotaxResetterMock()
```
Comment (Progi1984) [inline tests/Unit/Core/Tax/TaxOptionsConfigurationTest.php]:
```suggestion
    protected function createProductEcotaxResetterMock(): ProductEcotaxResetterInterface
```

**Review by Progi1984 — DISMISSED** [review]

**Comment by ghost** [comment]
Done @Progi1984 thank for review

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by HanaRebaiQA — CHANGES_REQUESTED** [review]
Hello @okom3pom 

I have tested this PR and i found this issue : 
The fields are not well aligned with checkboxes when a Shop configuration is selected
image
image
image
image
image

Could you please check it ?

Thanks!

**Comment by ghost** [comment]
Hello @HanaRebaiQA;

This is normal behaviour, or this problem is not related to this PR as it also does so in the maintenance page.

**Comment by NeOMakinG** [comment]
@HanaRebaiQA 

I think it's not blocking the PR, as https://github.com/PrestaShop/PrestaShop/pull/27430 will probably fix it when 1.7.8.x will be merged on develop!

**Review by HanaRebaiQA — APPROVED** [review]
Hello 

Thank you @NeOMakinG for your feedback!

@okom3pom 
So, the improvement is well done.

Here the verified cases : 
**-Check using different configurations on shops and groups of shops of multistore : OK**
https://watch.screencastify.com/v/1Xpee2N92DNBrUdv7vVN

Uploading PR27525.mp4…



**-Check with different langauges (RTL and NO RTL) : OK** 
5004b9c9-f5d1-4f74-86ea-23f17c89ad11
téléchargement
téléchargement (1)
téléchargement (2)


**-Check with different browsers (Chrome, Firefox) : OK**
image
image



**-Check with mobile : OK**
image

674d657b-e20a-4be0-8b6f-b50b572e8bd0
945a44a6-3c96-403b-b54f-64ab79d20178


**-Check with responsive mode : OK**
cc14ac9d-f4f0-4f54-83f1-692c0c2b416c
 

So, QA :heavy_check_mark: 

Thanks!

**Comment by Progi1984** [comment]
Thanks @okom3pom & @HanaRebaiQA

**Pull request merged**